game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/BattleStepsTest.java

```diff
@@ -122,8 +123,8 @@ public static UnitAndAttachment newUnitAndAttachment() {
     final Unit unit = mock(Unit.class);
     final UnitType unitType = mock(UnitType.class);
     final UnitAttachment unitAttachment = mock(UnitAttachment.class);
-    when(unit.getType()).thenReturn(unitType);
-    when(unitType.getAttachment(UNIT_ATTACHMENT_NAME)).thenReturn(unitAttachment);
+    lenient().when(unit.getType()).thenReturn(unitType);
```

Comment:
Because of the change in how the sub steps are figured out, these mocks are being called sometimes when they aren't needed.  But to work around that would require making the test methods more complicated.  I felt adding lenient() to it was ok.